2023/Q2/artifacts/PoS-quint/namada.qnt

```diff
@@ -187,7 +187,7 @@ module namada {
       if (amount <= delegatorState.balance and amount > 0) {
         val updatedDelegatorState = delegatorState.with("balance", delegatorState.balance - amount)
                                                   .with("bonded", delegatorState.bonded.set(validatorState.address,
-                                                                                            delegatorState.bonded.get(validatorState.address).setAdd({start: posState.epoch + PIPELINE_OFFSET, amount: amount})))                                                                       
+                                                                                            delegatorState.bonded.get(validatorState.address).setAdd({start: posState.epoch + PIPELINE_OFFSET, amount: amount}))) // is this always adding a new bond element rather than updating? maybe this doesn't matter if only 1 tx is permitted, but from a spec perspective not sure if this is incorrect.                                                                       
```

Comment:
yeah, I think this is related to the comment on:

```quint
    // Transactions per epoch
    // the spec is not fully ready to handle multiple txs per epoch
    // we would need to handle duplicate bonds and unbonds
    pure val TXS_EPOCH: int = 1
```

I think we could probably change `bonded: Address -> Set[Bond]` to `bonded: Address -> Epoch -> int` to handle that

Reply:
Exactly, https://github.com/informalsystems/partnership-heliax/pull/63 addresses it.